wrappers/flutter/lib/src/infra/fitatu_barcode_scanner_preview.dart

```diff
@@ -42,41 +30,45 @@ class FitatuBarcodeScannerPreviewState extends State<FitatuBarcodeScannerPreview
   @override
   void initState() {
     super.initState();
-
     SystemChrome.setPreferredOrientations([DeviceOrientation.portraitUp, DeviceOrientation.portraitDown]);
   }
 
   @override
   Widget build(BuildContext context) {
     late Widget preview;
 
-    Widget getCommonScanner() => CommonFitatuScannerPreview(
-      key: _key,
-      onResult: widget.onResult,
-      options: widget.options,
-      onChanged: widget.onChanged,
-      onError: widget.onError,
-      overlayBuilder: widget.previewOverlayBuilder,
-    );
+    return FutureBuilder(
+      future: widget.controllerLease.resource,
+      builder: (context, snapshot) {
+        final controller = snapshot.data;
 
-    if (widget.alwaysUseCommon || kIsWeb) {
-      preview = getCommonScanner();
-    } else if (Platform.isAndroid) {
-      preview = CameraPermissionsGuard(
-        child: AndroidFitatuScannerPreview(
-          key: _key,
-          onResult: widget.onResult,
-          options: widget.options,
-          onChanged: widget.onChanged,
-          onError: widget.onError,
-          overlayBuilder: widget.previewOverlayBuilder,
-        ),
-      );
-    } else {
-      preview = getCommonScanner();
-    }
+        if (Platform.isAndroid) {
+          preview = CameraPermissionsGuard(
+            child: _AndroidFitatuScannerPreview(
+              key: _key,
+              controller: controller != null ? controller as _AndroidBarcodeScanner : null,
```

Comment:
```suggestion
              controller: controller is _AndroidBarcodeScanner ? controller as _AndroidBarcodeScanner : null,
```

W teorii analogicznie w IOS